src/scripts/clientUtils.js

```diff
@@ -1,43 +1,56 @@
-import AppInfo from '../components/AppInfo';
 import ServerConnections from '../components/ServerConnections';
 import toast from '../components/toast/toast';
 import loading from '../components/loading/loading';
 import { appRouter } from '../components/appRouter';
 import baseAlert from '../components/alert';
 import baseConfirm from '../components/confirm/confirm';
 import globalize from '../scripts/globalize';
+import * as webSettings from './settings/webSettings';
 
 export function getCurrentUser() {
     return window.ApiClient.getCurrentUser(false);
 }
 
-//TODO: investigate url prefix support for serverAddress function
-export function serverAddress() {
-    if (AppInfo.isNativeApp) {
-        const apiClient = window.ApiClient;
-
-        if (apiClient) {
-            return apiClient.serverAddress();
-        }
+// TODO: investigate url prefix support for serverAddress function
+export async function serverAddress() {
+    const apiClient = window.ApiClient;
 
-        return null;
+    if (apiClient) {
+        return Promise.resolve(apiClient.serverAddress());
     }
 
-    const urlLower = window.location.href.toLowerCase();
-    const index = urlLower.lastIndexOf('/web');
+    const current = await ServerConnections.getAvailableServers().then(servers => {
+        if (servers.length !== 0) {
+            return Promise.resolve(servers[0].ManualAddress);
+        }
+    });
 
-    if (index != -1) {
-        return urlLower.substring(0, index);
-    }
+    // TODO this makes things faster but it also blocks the wizard in some scenarios
+    // if (current) return Promise.resolve(current);
 
-    const loc = window.location;
-    let address = loc.protocol + '//' + loc.hostname;
+    const urls = [];
+    urls.push(window.location.origin);
+    urls.push(`https://${window.location.hostname}:8920`);
+    urls.push(`http://${window.location.hostname}:8096`);
+    urls.push(...await webSettings.getServers());
 
-    if (loc.port) {
-        address += ':' + loc.port;
-    }
+    const promises = urls.map(url => {
+        return fetch(`${url}/System/Info/Public`).catch(error => {
```

Comment:
If `url` returns a valid response, then `url` is correct. However, `LocalAddress` is not trustworthy.

So it would be wise to return `url` if this fetch succeeds.

Reply:
This fixes my complaint

```patch
--- a/src/scripts/clientUtils.js
+++ b/src/scripts/clientUtils.js
@@ -35,18 +35,14 @@ export async function serverAddress() {
     urls.push(...await webSettings.getServers());

     const promises = urls.map(url => {
-        return fetch(`${url}/System/Info/Public`).catch(error => {
+        return fetch(`${url}/System/Info/Public`).then(resp => url).catch(error => {
             return Promise.resolve();
         });
     });

     return Promise.all(promises).then(responses => {
-        responses = responses.filter(response => response && response.ok);
-        return Promise.all(responses.map(response => response.json()));
-    }).then(configs => {
-        let selection = configs.find(config => !config.StartupWizardCompleted);
-        if (!selection) selection = configs[0];
-        return Promise.resolve(selection.LocalAddress);
+        responses = responses.filter(response => response);
+        return responses[0];
     }).catch(error => {
         console.log(error);
         return Promise.resolve();
```